### PR TITLE
Prevent exception from being raised when modifying dir permissions for clean up.

### PR DESCRIPTION
**Pull request by DailyDreaming**
I don't think the workflow should break if everything was successful but we can't clean up.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Prevent exception from being raised when modifying dir permissions for clean up.

## Reviewer Checklist

 * [x] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [x] Read through the code changes. Make sure that it doesn't have:
    * [x] Addition of trailing whitespace.
    * [x] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [x] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [x] New functions or classes without informative docstrings.
    * [x] Changes to semantics not reflected in the relevant docstrings.
    * [x] New or changed command line options for Toil workflows that are not reflected in `docs/running/{cliOptions,cwl,wdl}.rst` 
    * [x] New features without tests.
* [x] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [x] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/worker.py [hunk 1]
```diff
@@ -609,10 +609,13 @@ def make_parent_writable(func: Callable[[str], Any], path: str, _: Any) -> None:
             up after itself.
             """
             # Just chmod it for rwx for user. This can't work anyway if it isn't ours.
-            os.chmod(os.path.dirname(path),  stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+            try:
+                os.chmod(os.path.dirname(path),  stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+            except PermissionError:
+                pass
```
Comment (mr-c) [inline src/toil/worker.py:615]: Can there at least be a log entry?

Comment (adamnovak) [inline src/toil/worker.py:615]: Yeah, @mr-c is right; this should `logger.error` a complaint.

**Review by adamnovak — CHANGES_REQUESTED** [review]
We still want to know when cleanup doesn't work, but you're right it probably shouldn't fail the workflow.

Now someone will complain that a bunch of ostensibly successful workflows filled their disk. But the root cause will be that they're running in a weird environment with a recalcitrant filesystem and they need to do manual cleanup anyway, and they wouldn't have been able to swap the filesystem if the workflows had failed to complain about it. So it is probably better this way.

**Review by adamnovak — APPROVED** [review]
I just went and added the log entry I wanted.

**Review by mr-c — APPROVED** [review]

**Pull request merged**